baseplate/lib/circuit_breaker/factory.py

```diff
@@ -0,0 +1,164 @@
+import logging
+
+from contextlib import contextmanager
+from datetime import timedelta
+from typing import Any
+from typing import Dict
+from typing import Iterator
+from typing import Tuple
+from typing import Type
+
+from baseplate import Span
+from baseplate.clients import ContextFactory
+from baseplate.lib import config
+from baseplate.lib.circuit_breaker.breaker import Breaker
+from baseplate.lib.circuit_breaker.breaker import BreakerState
+from baseplate.lib.circuit_breaker.errors import BreakerTrippedError
+
+
+logger = logging.getLogger(__name__)
+
+
+class CircuitBreakerClientWrapperFactory(ContextFactory):
```

Comment:
Should you also be exposing a `report_runtime_metrics` that calls one on the `client_factory` if it exists? Or is the idea that you would still attach the bare client factory to the context?